std/algorithm/sorting.d

```diff
@@ -1048,147 +1048,287 @@ unittest
         @property auto save()
         {
-            auto ret = this;
-            foreach (ti, elem; _r)
+            auto result = this;
+            foreach (i, _; Rs)
             {
-                ret._r[ti] = elem.save;
+                result.source[i] = result.source[i].save;
             }
-            return ret;
+            return result;
         }
     }
 
     static if (allSatisfy!(hasLength, Rs))
     {
-        ///
         @property size_t length()
         {
             size_t result;
-            foreach (i, U; Rs)
+            foreach (i, _; Rs)
             {
-                result += _r[i].length;
+                result += source[i].length;
             }
             return result;
         }
 
-        ///
         alias opDollar = length;
     }
+
+    public Rs source;
+    private size_t _lastFrontIndex = size_t.max;
+    static if (isBidirectional)
+    {
+        private size_t _lastBackIndex = size_t.max;
+    }
 }
 
-/// Ditto
-Merge!(less, Rs) merge(alias less = "a < b", Rs...)
-(Rs rs)
+/**
+   Merge multiple sorted ranges `rs` with less-than predicate function `pred`
+   into one single sorted output range containing the sorted union of the
+   elements of inputs. Duplicates are not eliminated, meaning that the total
+   number of elements in the output is the sum of all elements in the ranges
+   passed to it; the `length` member is offered if all inputs also have
+   `length`. The element types of all the inputs must have a common type
+   `CommonType`.
+
+Params:
+    less = Predicate the given ranges are sorted by.
+    rs = The ranges to compute the union for.
+
+Returns:
+    A range containing the union of the given ranges.
+
+Details:
+
+All of its inputs are assumed to be sorted. This can mean that inputs are
+   instances of $(REF SortedRange, std,range). Use the result of $(REF sort,
+   std,algorithm,sorting), or $(REF assumeSorted, std,range) to merge ranges
+   known to be sorted (show in the example below). Note that there is currently
+   no way of ensuring that two or more instances of $(REF SortedRange,
+   std,range) are sorted using a specific comparison function `pred`. Therefore
+   no checking is done here to assure that all inputs `rs` are instances of
+   $(REF SortedRange, std,range).
+
+   This algorithm is lazy, doing work progressively as elements are pulled off
+   the result.
+
+   Time complexity is proportional to the sum of element counts over all inputs.
+
+   If all inputs have the same element type and offer it by `ref`, output
+   becomes a range with mutable `front` (and `back` where appropriate) that
+   reflects in the original inputs.
+
+   If any of the inputs `rs` is infinite so is the merge result (`empty` is
+   always `false`).
+
+   Example:
+   -------
+   auto a = [0, 2, 4];
+   auto b = [1, 3, 5];
+   auto c = merge(a.assumeSorted, b.assumeSorted)
+   assert(equal(c, [0, 1, 2, 3, 4, 5]));
+   assert(equal(c.retro, [5, 4, 3, 2, 1, 0]));
+   -------
+*/
```

Comment:
Oh, that's because it's been deprecated in favor of `merge`. I guess I was missing something after all. Never mind :)